### PR TITLE
Application Deploy: Support deploy of Git apps

### PR DESCRIPTION
**Pull request by irfanhabib**

### CODE REVIEW

File: src/app/store/effects/deploy-app.effects.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  ) {
+  }
+
+  @Effect() checkAppExists$ =  this.actions$.ofType<CheckProjectExists>(CHECK_PROJECT_EXISTS)
+  .withLatestFrom(this.store.select(selectDeployAppState))
+  .filter(([action, state]) => {
+   return state.projectExists && state.projectExists.checking;
+  })
+  .switchMap(([action, state]: any) => {
+      return this.http.get(`https://api.github.com/repos/${action.projectName}`)
```
Comment (richard-cox) [inline src/app/store/effects/deploy-app.effects.ts]: We should handle fetching requests/entities via the generic framework

File: src/app/store/effects/deploy-app.effects.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+          return [new ProjectExists(action.projectName, res), new DeleteCachedBranches()];
+        })
+        .catch(err => {
+          return Observable.of(new ProjectDoesntExist(action.projectName));
+        });
+    });
+
+  @Effect() fetchBranches$ =  this.actions$.ofType<FetchBranchesForProject>(FETCH_BRANCHES_FOR_PROJECT)
+  .switchMap((action: any) => {
+      return this.http.get(`https://api.github.com/repos/${action.projectName}/branches`)
```
Comment (richard-cox) [inline src/app/store/effects/deploy-app.effects.ts]: Same here

File: src/app/store/effects/deploy-app.effects.ts (hunk 1 above)
Comment (irfanhabib) [reply src/app/store/effects/deploy-app.effects.ts]: I've migrated the Branch and Commits effects to use the generic framework. This effect is used as part of a filter and doesn't need to store any entity.

**Comment by irfanhabib** [comment]
@richard-cox I've addressed the comments and updated the PR

**Comment by richard-cox** [comment]
Tracked other issues in https://github.com/SUSE/stratos/issues/200 and https://github.com/SUSE/stratos/issues/201

**Pull request merged**